Makefile

```diff
@@ -1,102 +1,170 @@
+
+.PHONY: build-release
+build-release: venv ## Build a faster version of binaries
+	@unset CONDA_PREFIX && cd python/pecos-rslib/ && $(VENV_BIN)/maturin develop --release
+	@unset CONDA_PREFIX && cd python/quantum-pecos && $(VENV_BIN)/pip install -e .[all]
+
+.PHONY: build-native
+build-native: venv ## Build a faster version of binaries with native CPU optimization
+	@unset CONDA_PREFIX && cd python/pecos-rslib/ && RUSTFLAGS='-C target-cpu=native' \
+	&& $(VENV_BIN)/maturin develop --release
+	@unset CONDA_PREFIX && cd python/quantum-pecos && $(VENV_BIN)/pip install -e .[all]
 
 # Documentation
 # -------------
 
-docs: install-all  ## Generate documentation
-	$(VENV_BIN)/pip install -r ./docs/requirements.txt
-	$(MAKE) -C docs SPHINXBUILD=../$(VENV_BIN)/sphinx-build clean html
+.PHONY: docs
+docs:  ## Generate documentation
+	#TODO: ...
 
 # Linting / formatting
 # --------------------
 
-lint: metadeps  ## Run all quality checks / linting / reformatting
+.PHONY: check
+check:  ## Run cargo check with all features
+	cargo check --workspace --all-targets --all-features
+
+.PHONY: clippy
+clippy:  ## Run cargo clippy with all features
+	cargo clippy --workspace --all-targets --all-features -- -D warnings
+
+.PHONY: fmt
+fmt: ## Run autoformatting for cargo
+	cargo fmt --all -- --check
+
+.PHONY: pre-commit  ## Run all quality checks / linting / reformatting
+pre-commit: venv fmt clippy
 	$(VENV_BIN)/pre-commit run --all-files
 
 # Testing
 # -------
 
-tests: install metadeps  ## Run tests on the Python package (not including optional dependencies)
-	$(VENV_BIN)/pytest tests -m "not optional_dependency"
+.PHONY: rstest
+rstest:  ## Run Rust tests
+	cargo test
 
-tests-dep: install-all metadeps ## Run tests on the Python package only for optional dependencies
-	$(VENV_BIN)/pytest tests -m optional_dependency
+.PHONY: pytest
+pytest:  ## Run tests on the Python package (not including optional dependencies). ASSUMES: previous build command
+	@cd python/tests/ && $(VENV_BIN)/pytest . -m "not optional_dependency"
 
-doctests:  ## Run doctests with pytest
-	$(VENV_BIN)/pytest ./docs --doctest-glob=*.rst --doctest-continue-on-failure
+.PHONY: pytest-dep
+pytest-dep: ## Run tests on the Python package only for optional dependencies. ASSUMES: previous build command
+	@cd python/ && $(VENV_BIN)/pytest tests -m optional_dependency
 
-tests-all: tests tests-dep doctests ## Run all tests
+.PHONY: pytest-doc
+pydoctest:  ## Run doctests with pytest. ASSUMES: A build command was ran previously. ASSUMES: previous build command
+	# TODO: update and install docs requirements
+	@cd python/ && $(VENV_BIN)/pytest docs --doctest-glob=*.rst --doctest-continue-on-failure
 
-# Building / Developing
-# ---------------------
-
-clean:  ## Clean up caches and build artifacts
-	-rm -rf *.egg-info dist build docs/_build .pytest_cache/ .ruff_cache/
+.PHONY: test-all
+test-all: rstest pytest pytest-dep pydoctest ## Run all tests. ASSUMES: previous build command
```

Comment:
I'm getting a lot errors of this `pecos is not a package` form: 

```bash
___________________________________________ ERROR collecting pecos/unit/test_blocks.py ____________________________________________
ImportError while importing test module '/home/peter/Downloads/PECOS-feat-rust/python/tests/pecos/unit/test_blocks.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
pecos/unit/test_blocks.py:13: in <module>
    from pecos.classical_interpreters.phir_classical_interpreter import (
E   ModuleNotFoundError: No module named 'pecos.classical_interpreters'; 'pecos' is not a package
```

Full test failure list:
```bash
ERROR pecos/integration/example_tests/test_basic_logical_sim.py
ERROR pecos/integration/example_tests/test_finding_threshold.py
ERROR pecos/integration/example_tests/test_recovery.py
ERROR pecos/integration/state_sim_tests/test_cointoss.py
ERROR pecos/integration/state_sim_tests/test_stab_sims/test_gate_init.py
ERROR pecos/integration/state_sim_tests/test_stab_sims/test_gate_one_qubit.py
ERROR pecos/integration/state_sim_tests/test_stab_sims/test_gate_two_qubit.py
ERROR pecos/integration/state_sim_tests/test_statevec.py
ERROR pecos/integration/test_phir.py
ERROR pecos/integration/test_phir_64_bit.py
ERROR pecos/integration/test_phir_setting_cregs.py
ERROR pecos/integration/test_quantum_circuits.py
ERROR pecos/integration/test_random_circuits.py
ERROR pecos/regression/test_qasm/examples/test_logical_steane_code_program.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_measures.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_preps.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_rots.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_sq_face_rots.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_sq_hadamards.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_sq_noncliffords.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_sq_paulis.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_sq_sqrt_paulis.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_tq_cliffords.py
ERROR pecos/regression/test_qasm/pecos/qeclib/qubit/test_tq_noncliffords.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/decoders/test_lookup.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/gates_sq/test_face_rots.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/gates_sq/test_hadamards.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/gates_sq/test_paulis.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/gates_sq/test_sqrt_paulis.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/gates_tq/test_transversal_tq.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/meas/test_destructive_meas.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/meas/test_measure_x.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/meas/test_measure_z.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/preps/test_encoding_circ.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/preps/test_pauli_states.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/preps/test_plus_h_state.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/preps/test_t_plus_state.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/qec/test_qec_3parallel.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/syn_extract/test_six_check_nonflagging.py
ERROR pecos/regression/test_qasm/pecos/qeclib/steane/syn_extract/test_three_parallel_flagging.py
ERROR pecos/regression/test_qasm/random_cases/test_control_flow.py
ERROR pecos/regression/test_qasm/random_cases/test_slr_phys.py
ERROR pecos/unit/reps/pymir/test_name_resolver.py
ERROR pecos/unit/test_binarray.py
ERROR pecos/unit/test_blocks.py
```

Reply:
Did you follow this note? ## Run all tests. ASSUMES: previous build command

Although, if you are having issues... probably need to find another solution. :D

Reply:
Weird. Try again please 🙂